cmd/app/main.go

```diff
@@ -44,7 +48,43 @@ func main() {
 		IdleTimeout:  appConf.Server.TimeoutIdle,
 	}
 
-	if err := s.ListenAndServe(); err != nil && err != http.ErrServerClosed {
-		logger.Fatal().Err(err).Msg("Server startup failed")
+	// Make a channel to listen for errors coming from the listener. Use a buffered
+	// channel so the goroutine exits if we dont get this error
+	serverErrors := make(chan error, 1)
+
+	// Start server in a goroutine so that its does not block
+	go func() {
+		logger.Info().Msgf("main :  API listening on %s.", s.Addr)
+		serverErrors <- s.ListenAndServe()
+	}()
+
+	// Make a channel to listen for an interrupt or terminate signal from the OS.
+	shutdown := make(chan os.Signal, 1)
+	signal.Notify(shutdown, os.Interrupt, os.Kill)
+
+	// Block main until signal is received
+	select {
+	case err := <-serverErrors:
+		logger.Info().Msgf("error : listening and serving %s", err)
+
+	case <-shutdown:
+		logger.Info().Msg("main : Start shutdown")
+
+		// Give outstanding requests a deadline for completion
+		const timeout = 30 * time.Second
```

Comment:
30s should be in appConf.Server. May be you can use appConf.Server.TimeoutIdle directly.